src/client/Microsoft.Identity.Client/Platforms/Android/Broker/AndroidAccountManagerBroker.cs

```diff
@@ -0,0 +1,452 @@
+            {
+                _logger.Verbose("User is specified for silent token request. Starting silent Android broker request. ");
+                string silentResult = await GetBrokerAuthTokenSilentlyAsync(brokerRequest, _parentActivity).ConfigureAwait(false);
+                return _brokerHelper.HandleSilentAuthenticationResult(silentResult, brokerRequest.CorrelationId);
+            }
+        }
+
+        private async Task<string> GetBrokerAuthTokenSilentlyAsync(BrokerRequest brokerRequest, Activity callerActivity)
+        {
+            brokerRequest = UpdateRequestWithAccountInfo(brokerRequest, callerActivity);
+            Bundle silentOperationBundle = _brokerHelper.CreateSilentBrokerBundle(brokerRequest);
+            silentOperationBundle.PutString(BrokerConstants.BrokerAccountManagerOperationKey, BrokerConstants.AcquireTokenSilent);
+
+            return await AcquireTokenSilentInternalAsync(silentOperationBundle).ConfigureAwait(false);
+        }
+
+        private async Task<string> AcquireTokenSilentInternalAsync(Bundle silentOperationBundle)
+        {
+            Bundle accountManagerResult = await PerformAccountManagerOperationAsync(silentOperationBundle).ConfigureAwait(false);
+
+            if (accountManagerResult != null)
+            {
+                return _brokerHelper.GetSilentResultFromBundle(accountManagerResult);
+            }
+
+            _logger.Info("Android Broker didn't return any results. ");
+            return null;
+        }
+
+        /// <summary>
+        /// This method is only used for Silent authentication requests so that we can check to see if an account exists in the account manager before
+        /// sending the silent request to the broker. 
+        /// </summary>
+        private BrokerRequest UpdateRequestWithAccountInfo(BrokerRequest brokerRequest, Activity callerActivity)
+        {
+            var accounts = GetBrokerAccounts(brokerRequest, callerActivity);
+
+            if (string.IsNullOrEmpty(accounts))
+            {
+                _logger.Info("Android account manager didn't return any accounts. ");
+                throw new MsalUiRequiredException(MsalError.NoAndroidBrokerAccountFound, MsalErrorMessage.NoAndroidBrokerAccountFound);
+            }
+
+            return _brokerHelper.UpdateBrokerRequestWithAccountData(accounts, brokerRequest);
+        }
+
+        private string GetBrokerAccounts(BrokerRequest brokerRequest, Activity callerActivity)
+        {
+            Bundle getAccountsBundle = _brokerHelper.CreateBrokerAccountBundle(brokerRequest);
+            getAccountsBundle.PutString(BrokerConstants.BrokerAccountManagerOperationKey, BrokerConstants.GetAccounts);
+
+            //This operation will acquire all of the accounts in the account manager for the given client ID
+            Bundle bundleResult = PerformAccountManagerOperationAsync(getAccountsBundle).Result;
+            return bundleResult?.GetString(BrokerConstants.BrokerAccounts);
```

Comment:
we already did this in `InitiateBrokerHandshakeAysnc()` on line 96ish...is there a way to only need to call `PerformAccountManagerOpAsync()` once?

Reply:
Not yet. Once we have handshake caching we can remove